core/object/configuration_info.h

```diff
@@ -0,0 +1,55 @@
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef CONFIGURATION_INFO_H
+#define CONFIGURATION_INFO_H
+
+#ifdef TOOLS_ENABLED
+
+#include "core/object/object.h"
+#include "core/string/ustring.h"
+#include "core/variant/dictionary.h"
+
+#define CONFIG_INFO(message) warnings.push_back(ConfigurationInfo::_make_configuration_info(message, "", "info"));
+#define CONFIG_INFO_P(message, property_name) warnings.push_back(ConfigurationInfo::_make_configuration_info(message, property_name, "info"));
+#define CONFIG_WARNING(message) warnings.push_back(ConfigurationInfo::_make_configuration_info(message, "", "warning"));
+#define CONFIG_WARNING_P(message, property_name) warnings.push_back(ConfigurationInfo::_make_configuration_info(message, property_name, "warning"));
+#define CONFIG_ERROR(message) warnings.push_back(ConfigurationInfo::_make_configuration_info(message, "", "error"));
+#define CONFIG_ERROR_P(message, property_name) warnings.push_back(ConfigurationInfo::_make_configuration_info(message, property_name, "error"));
```

Comment:
I'd remove unused macros for now. They are simple enough to add and can be added as needed.

Reply:
Alright, keeping only the warning macros. And yeah not sure the variable rename is worth the effort, so I'd keep it as-is for now.